code/modules/mob/inventory_procs.dm

```diff
@@ -40,6 +40,13 @@
 	if(istype(r_hand, typepath))
 		return r_hand
 
+// List version of above proc
+// Returns ret_item, which is either the successfully located item or null
+/mob/proc/is_holding_item_of_types(list/typepaths)
+	for(var/typepath in typepaths)
+		var/ret_item = is_holding_item_of_type(typepath)
+		return ret_item
+
```

Comment:
I don't think this does what you want it to do -- it'll return on the first element in the list regardless. I think you might only want to return if ret_item is true/not null